src/components/CartCard.jsx

```diff
@@ -0,0 +1,101 @@
+//ToDo Rename this file, confused in the structure a bit
+//ToDo make UI Look good. 
+
+import axios from "axios";
+import { useState } from "react";
+import useMainState from "../contexts/MainContextProvider";
+import ENUMS from '../utils/enums';
+
+const { SET_CART, SET_WISHLIST } = ENUMS;
+
+const SingleProductCart = ({ product, quantity }) => {
+    // 0 ->  not done, 1 -> loading, 2-> successful, -1 -> failed
+    const [addingToWishlist, setAddingToWishlist] = useState(0);
+    const [updatingCart, setUpdatingCart] = useState(false);
+    const {dispatch} = useMainState();
+
+
+    const updateCart = (quantity) => {
+        setUpdatingCart(true);
+        axios.post(`${process.env.REACT_APP_SERVER_URL}/cart/`, {
```

Comment:
create a global variable for process.env.REACT_APP_SERVER_URL and then use it here, it will be more readable then